__tests__/register.test.ts

```diff
@@ -0,0 +1,186 @@
+import {register, RegisterOptions} from '../src/register'
+import {terraform, TerraformOptions} from '../src/lib/terraform'
+
+import nock from 'nock'
+
+beforeEach(() => {
+  nock.cleanAll()
+})
+
+test('Should throw if the VCS client is not recognized', async () => {
+  const regOpts: RegisterOptions = {
+    organization: 'org',
+    repo: 'org/terraform-unit-test',
+    vcsName: 'not-github'
+  }
+  const tfOpts: TerraformOptions = {
+    token: '12345',
+    host: 'fake.terraform.io'
+  }
+
+  nock(`https://${tfOpts.host}`)
+    .get(`/api/v2/organizations/${regOpts.organization}/oauth-clients`)
+    .reply(200, {
+      data: [{attributes: {name: 'github'}}]
+    })
+
+  await expect(register(regOpts, terraform(tfOpts))).rejects.toMatchObject({
+    message: `VCS client "${regOpts.vcsName}" not found`
```

Comment:
It's better to hardcode stuff like this (slugs, names) and just pass the VCS options as a literal. In other words, repetition is justified in unit tests since it's easier to read the tests than if they use a bunch of dynamic strings in expectations.